packages/PHPStanStaticTypeMapper/TypeMapper/ClosureTypeMapper.php

```diff
@@ -39,16 +30,7 @@ public function getNodeClass(): string
      */
     public function mapToPHPStanPhpDocTypeNode(Type $type): TypeNode
     {
-        $identifierTypeNode = new IdentifierTypeNode($type->getClassName());
-
-        $returnDocTypeNode = $this->phpStanStaticTypeMapper->mapToPHPStanPhpDocTypeNode($type->getReturnType());
-
-        $callableTypeParameterNodes = $this->createCallableTypeParameterNodes($type);
-
-        // callable parameters must be of specific type
-        Assert::allIsInstanceOf($callableTypeParameterNodes, CallableTypeParameterNode::class);
-
-        return new SpacingAwareCallableTypeNode($identifierTypeNode, $callableTypeParameterNodes, $returnDocTypeNode);
+        return $type->toPhpDocNode();
```

Comment:
@TomasVotruba this may need fqcn with add `\` if Closure is not in use statement, as `mapToPhpParserNode()` returns `FullyQualified` object.

this fqcn closure is valid doc https://phpstan.org/r/f2a7c394-a8b0-45ec-aebb-f0ad7dabe95a

Reply:
https://github.com/rectorphp/rector-src/pull/4584